### PR TITLE
Test circleci workflow

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Automatically deploy any branch not created by violinist to acquia.
- Reduce sites tests by behat, but randomly choose which ones.
- Hopefully automatically deploy tags to acquia

# Needed By (Date)
- whenever

# Urgency
- low

# Steps to Test
1. review the circleci reports
1. optionally see the branch has been sent to [acquia dashboard](https://cloud.acquia.com/app/develop/applications/23a85077-2967-41a4-be22-a84c24e0f81a)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Review by boznik — APPROVED** [review]
This seems to be working as expected, but right as i said that the method switched from release-it to robo...

File: .circleci/RoboFile.php [hunk 1]
```diff
@@ -42,6 +51,165 @@ class RoboFile extends Tasks {
...
+
+    if (!$result->wasSuccessful()) {
+      throw new Exception('Release commit was unsuccessful');
+    }
+
+    // Create a new release in github. This will generate a tag which will be
+    // used in another CircleCI task.
+    $result = $this->taskGitHubRelease($version)
+      ->accessToken(getenv('GITHUB_TOKEN'))
+      ->uri('SU-HSDO/suhumsci')
```
Comment (sherakama) [inline .circleci/RoboFile.php]: Can this be configurable through blt.yml?

Comment (pookmish) [reply .circleci/RoboFile.php]: I'll parse the git remote url. using blt in this robo would require alot of inheritance

File: .circleci/RoboFile.php [hunk 2]
```diff
@@ -42,6 +51,165 @@ class RoboFile extends Tasks {
...
+    }
+  }
+
+  /**
+   * Get the last version from the profile.
+   *
+   * @return string
+   *   Last semver version.
+   */
+  protected function getLastVersion() {
```
Comment (sherakama) [inline .circleci/RoboFile.php:133]: So this locks the build version with the profile version? Perhaps we need a way to determine what the build version is as I don't think these will be 1-1 on every project.

Comment (pookmish) [reply .circleci/RoboFile.php:133]: I was grabbing the latest git tag, but i wasnt confident that was in semver

File: .circleci/RoboFile.php [hunk 3]
```diff
@@ -42,6 +51,165 @@ class RoboFile extends Tasks {
...
+
+  /**
+   * Set the versions in the module & profile info files.
+   *
+   * @param string $version
+   *   New version.
+   */
+  protected function setVersions($version) {
+    // Update only modules in humsci and the appropriate profiles.
+    $dirs = [
```
Comment (sherakama) [inline .circleci/RoboFile.php:211]: Oh interesting, you are also updating the 'custom' modules that are in this repo. 🤔 I like that it is automated, I'm not sure about the coupling of versions.

File: .circleci/RoboFile.php (hunk 2 above)
Comment (pookmish) [reply .circleci/RoboFile.php:133]: We could change this logic if we use this on other projects

File: .circleci/RoboFile.php (hunk 3 above)
Comment (pookmish) [reply .circleci/RoboFile.php:211]: i've been manually updating the humsci modules to be the same as the profile. :man_shrugging:

File: .circleci/RoboFile.php (hunk 1 above)
Comment (pookmish) [reply .circleci/RoboFile.php]: that should work better now

File: .circleci/RoboFile.php (hunk 2 above)
Comment (sherakama) [reply .circleci/RoboFile.php:133]: Aye, sorry, I've been doing review through the ACSF lens. I'll re-adjust my thinking and take another pass.

**Comment by pookmish** [comment]
Changin the base branch to `develop`

**Comment by pookmish** [comment]
Any feedback on this?

**Review by sherakama — APPROVED** [review]
I approve of it with the requirement that you demo it at the next developer chat.

**Comment by pookmish** [comment]
sweet

**Pull request merged**